src/Proto3/Suite/Class.hs

```diff
@@ -605,7 +607,7 @@ class Message a where
 
 -- | Generate metadata for a message type.
 message :: (Message a, Named a) => Proxy a -> DotProtoDefinition
-message pr = DotProtoMessage (Single $ nameOf pr) $ DotProtoMessageField <$> (dotProto pr)
+message = DotProtoMessage <$> (Single . nameOf) <*> (fmap DotProtoMessageField . dotProto)
```

Comment:
I think the original version was more clear.  My usual rule of thumb is that if a piece of code uses `Functor`/`Applicative`/`Monad` in more than one way then I usually monomorphize all but one of the uses.  For example, here you use `<$>`/`<*>` in one case for `Reader (Proxy a)` and `fmap` in another way for `[]`

Reply:
I like the pointful version you just suggested the most. That is easy for me to understand at first-glance.